packages/flutter/lib/src/widgets/stretch_effect.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'image_filter.dart';
+
+/// A widget that applies a stretching visual effect to its child.
+///
+/// When shader-based effects are supported, they are used for better performance
+/// and visual quality. Otherwise, a matrix transform is applied as a fallback.
+///
+/// Used by [StretchingOverscrollIndicator] widget.
+class StretchEffect extends StatelessWidget {
```

Comment:
Can you explain why this widget should be made public? In other words, do you expect this widget to be used independently outside of `StretchingOverscrollIndicator`?

Reply:
@MTtankkeo Hi can you answer this question?